sstables/sstables.cc

```diff
@@ -2163,31 +2163,31 @@ future<> sstable::check_create_links_replay(const sstring& dst_dir, generation_t
     sstlog.trace("create_links: {} -> {} generation={} mark_for_removal={}", get_filename(), dst_dir, generation, mark_for_removal);
     auto comps = all_components();
-        co_await check_create_links_replay(dst_dir, generation, comps);
-            // TemporaryTOC is always first, TOC is always last
-            auto dst = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, component_type::TemporaryTOC);
-            co_await sstable_write_io_check(idempotent_link_file, filename(component_type::TOC), std::move(dst));
-                co_await sstable_write_io_check(sync_directory, dst_dir);
-                co_await parallel_for_each(comps, [this, &dst_dir, generation] (auto p) {
-                    auto src = sstable::filename(_storage.dir, _schema->ks_name(), _schema->cf_name(), _version, _generation, _format, p.second);
-                    auto dst = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, p.second);
-                    return sstable_write_io_check(idempotent_link_file, std::move(src), std::move(dst));
-                });
-                co_await sstable_write_io_check(sync_directory, dst_dir);
-            auto dst_temp_toc = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, component_type::TemporaryTOC);
-            if (mark_for_removal) {
-                // Now that the source sstable is linked to new_dir, mark the source links for
-                // deletion by leaving a TemporaryTOC file in the source directory.
-                auto src_temp_toc = sstable::filename(_storage.dir, _schema->ks_name(), _schema->cf_name(), _version, _generation, _format, component_type::TemporaryTOC);
-                co_await sstable_write_io_check(rename_file, std::move(dst_temp_toc), std::move(src_temp_toc));
-                    co_await sstable_write_io_check(sync_directory, _storage.dir);
-            } else {
-                // Now that the source sstable is linked to dir, remove
-                // the TemporaryTOC file at the destination.
-                co_await sstable_write_io_check(remove_file, std::move(dst_temp_toc));
-            }
-            co_await sstable_write_io_check(sync_directory, dst_dir);
-            sstlog.trace("create_links: {} -> {} generation={}: done", get_filename(), dst_dir, generation);
+    co_await check_create_links_replay(dst_dir, generation, comps);
+    // TemporaryTOC is always first, TOC is always last
+    auto dst = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, component_type::TemporaryTOC);
+    co_await sstable_write_io_check(idempotent_link_file, filename(component_type::TOC), std::move(dst));
+    co_await sstable_write_io_check(sync_directory, dst_dir);
+    co_await parallel_for_each(comps, [this, &dst_dir, generation] (auto p) {
```

Comment:
nit: use coroutine::parallel_for_each

Reply:
(and by the way, I'm not sure how much creating the links in parallel buys us here, if at all...)

Reply:
the comment applies to the previous patch of course, not to the indentation fix

Reply:
(off-topic) On a multi-commits PR like this, given a patch comment to a specific hunk, how do I know which patch it belongs to?

Reply:
Thanks to github.com, you can't tell, AFAIK.
(and I hope they don't ban me for saying that, well not until a new billionaire owner acquires them... ;-))

Reply:
If you go to the patch list and scroll through ('n'), you'll eventually see it in context. Of course it's not fun with 37 patches.

Reply:
> (and by the way, I'm not sure how much creating the links in parallel buys us here, if at all...)

It changes the latency from 0.5ms * nr_links to 0.5ms, since the syscall thread will be able to issue them without waiting. Even better with io_uring, once we forward renames to the ring.